features/dd-sdk-android-trace-internal/src/main/java/com/datadog/trace/core/propagation/HttpCodec.java

```diff
@@ -363,4 +365,16 @@ static String firstHeaderValue(final String value) {
     int firstComma = value.indexOf(',');
     return firstComma == -1 ? value : value.substring(0, firstComma).trim();
   }
+
+  @NonNull
+  static Baggage composeBaggage(@NonNull DDSpanContext context) {
+    Baggage baggage = new Baggage();
+    final String sessionId = (String) context.getTags().get(RUM_SESSION_ID_KEY);
```

Comment:
shouldn't we take everything from the `context`? because if it is extracted context and there were other tags there, we won't propagate them

Reply:
Not clear what "everything" we should take from the context. This code just extraction from existing logic in `DatadogHttpCodec` and `W3CHttpCodec. 
Are you suggesting to get `sessionId` directly from context, but not from the `tags` ?

Reply:
Nevermind. I was thinking that this baggage should go into `ExtractedContext` if we are working with the request already having propagation context, but it seems Codecs logic of working with `baggageItems` is about `ot-baggage` (so I was thinking that proper merge should be done in the `inject` as well, preserving items from `DDSpanContext`), but it seems now the work of supporting this is shifted to `mergeBaggage` in `TracingInterceptor`.